action/tx_legacy.go

```diff
@@ -19,22 +19,24 @@ var _ TxCommonInternal = (*LegacyTx)(nil)
 // LegacyTx is the legacy transaction
 type LegacyTx struct {
 	chainID  uint32
+	version  uint32
 	nonce    uint64
 	gasLimit uint64
 	gasPrice *big.Int
 }
 
 // NewLegacyTx creates a new legacy transaction
-func NewLegacyTx(chainID uint32, nonce uint64, gasLimit uint64, gasPrice *big.Int) *LegacyTx {
+func NewLegacyTx(chainID, version uint32, nonce uint64, gasLimit uint64, gasPrice *big.Int) *LegacyTx {
```

Comment:
The `version` parameter should not be needed in `NewLegacyTx`; the versions we create should all be version=0, and versions not equal to 0 should be set through `fromProto`.

Reply:
good point